toolprovider/mise/install_plugin.go

```diff
@@ -0,0 +1,139 @@
+package mise
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/bitrise-io/bitrise/v2/log"
+	"github.com/bitrise-io/bitrise/v2/toolprovider/provider"
+)
+
+type PluginSource struct {
+	PluginName  provider.ToolID
+	GitCloneURL string
+}
+
+var pluginSourceMap = map[provider.ToolID]PluginSource{
+	"flutter": {PluginName: "flutter", GitCloneURL: "https://github.com/asdf-community/asdf-flutter.git"},
+	"tuist":   {PluginName: "tuist", GitCloneURL: "https://github.com/tuist/asdf-tuist.git"},
+}
+
+var miseCoreTools = []string{
+	"bun",
+	"deno",
+	"elixir",
+	"erlang",
+	"go",
+	"golang",
+	"java",
+	"node",
+	"nodejs",
+	"python",
+	"ruby",
+	"rust",
+	"swift",
+	"zig",
+}
+
+// InstallPlugin installs a plugin for the specified tool, if needed.
+//
+// It resolves the plugin source from the tool request or predefined map,
+// checks if the plugin is already installed, and if not, installs it using mise.
+func (m *MiseToolProvider) InstallPlugin(tool provider.ToolRequest) error {
+	if slices.Contains(miseCoreTools, string(tool.ToolName)) {
+		// Core tools do not require plugin installation.
```

Comment:
This is true, but are you sure we want to skip the install if the user explicitly provides a plugin URL?

[This page](https://mise.jdx.dev/core-tools.html) says the core tools can be overridden with an explicit plugin install